extensions/identity-hub-store-spi/src/main/java/org/eclipse/dataspaceconnector/identityhub/store/IdentityHubStore.java

```diff
@@ -12,12 +12,17 @@
  *
  */
 
-package org.eclipse.dataspaceconnector.dtos;
+package org.eclipse.dataspaceconnector.identityhub.store;
+
+import java.util.Collection;
 
 /**
- * Web Node interface
- * See: <a href="https://identity.foundation/decentralized-web-node/spec/#interfaces">Web node interface identity foundation documentation.</a>
+ * IdentityHubStore store used to store HubObjects.
+ * A custom IdentityHubStore store can be injected via service loader as an EDC Service extension.
  */
-// TODO: Implement interface data model https://identity.foundation/decentralized-web-node/spec/#interfaces
-public class WebNodeInterface {
+public interface IdentityHubStore<T> {
```

Comment:
why the generic type? This has no particular support by EDC, and seems confusing

Reply:
Why not? The API gives freedom to restrict the type of the data that can be handled by the store.

Reply:
Because of type erasure you can't restrict the type which will be injected. So anyway you will only have one instance and one value for T. What would be the downside or removing T and handling `Object`s?